src/Microsoft.AspNetCore.Authentication.Cookies/CookieAuthenticationHandler.cs

```diff
@@ -244,21 +247,34 @@ private CookieOptions BuildCookieOptions()
             }
         }
 
-        protected override async Task HandleSignInAsync(ClaimsPrincipal user, AuthenticationProperties properties)
+        public async Task SignInAsync(ClaimsPrincipal user, AuthenticationProperties properties)
+        {
+            if (user == null)
+            {
+                throw new ArgumentNullException(nameof(user));
+            }
+
+            properties = properties ?? new AuthenticationProperties();
+            await HandleSignInAsync(user, properties);
+            Logger.SignedIn(Scheme.Name);
+        }
+
+        protected virtual async Task HandleSignInAsync(ClaimsPrincipal user, AuthenticationProperties properties)
```

Comment:
No I just copied the old methods over, I'll collapse this into the single public virtual SignIn